cmd/buildah/bud.go

```diff
@@ -105,12 +119,38 @@ func budCmd(c *cobra.Command, inputArgs []string, iopts budResults) error {
 		pullPolicy = imagebuildah.PullAlways
 	}
 
-	args := make(map[string]string)
+	args := make(argArray)
+
+	p := platforms.DefaultSpec()
+	if c.Flag("platform").Changed {
+		var err error
+		p, err = platforms.Parse(iopts.Platform)
+		if err != nil {
+			return errors.Wrapf(err, "error parsing --platform argument")
+		}
+		args.setArch(p)
+	} else {
+		args.setArch(platforms.DefaultSpec())
```

Comment:
Why not us p?  Remove the else {}

args.setArch(p)

Reply:
doh, of course. Fixing.